draft-ietf-quic-transport.md

```diff
@@ -575,47 +627,41 @@ QUIC's connection establishment begins with version negotiation, since all
 communication between the endpoints, including packet and frame formats, relies
 on the two endpoints agreeing on a version.
 
-A QUIC connection begins with a client sending a handshake packet. The details
-of the handshake mechanisms are described in {{handshake}}, but all of the
-initial packets sent from the client to the server MUST have the VERSION flag
-set, and MUST specify the version of the protocol being used.
+A QUIC connection begins with a client sending a handshake packet. Until the
+cryptographic handshake ({{handshake}}) produces 1-RTT keys, packets sent by a
+client MUST include the version in the packet header.  This allows the server to
+identify early packets and to enable version negotiation.
```

Comment:
We had a long discussion about that in relation to the TLS draft.  Most relevant is this comment from Dragana: https://github.com/quicwg/base-drafts/pull/43#discussion_r91596541  If I don't make this change, it isn't possible for the packet protection to work out which keys to use on a given packet.

Yes, this is different from Google QUIC, but that's not a downside.  Google QUIC relied very heavily on trial decryption.  This change would mean no need for trial decryption at all.